File: solutions/Leetcode_1423/leetcode_1423.py
from typing import List


class Solution1:
    def max_score(self, cardPoints: List[int], k: int) -> int:
        length = len(cardPoints)
        if k >= length:
            return sum(cardPoints)
        maximum_score = score = sum(cardPoints[:k])
        for i in range(-1, -k-1, -1):
            score = score - cardPoints[k+i] + cardPoints[i]
            maximum_score = max(maximum_score, score)
        return maximum_score


if __name__ == '__main__':
    nums = [1, 100, 1]
    k = 1
    ans = Solution1().max_score(nums, k)
    print(ans)

File: solutions/Leetcode_1423/leetcode_1423.cc
#include <numeric>
#include <vector>

using namespace std;

class Solution1
{
public:
    int maxScore(vector<int> cardPoints, int k)
    {
        int length = cardPoints.size();
        if (k >= length)
            return accumulate(cardPoints.begin(), cardPoints.end(), 0);
        int score = accumulate(cardPoints.begin(), cardPoints.begin() + k, 0), max_score = score;
        for (int i = 1; i <= k; i++)
        {
            score = score - cardPoints[k - i] + cardPoints[length - i];
            max_score = max(score, max_score);
        }
        return max_score;
    }
};